all/HOPE-3--Dependency-Injection.md

```diff
@@ -23,10 +23,10 @@ This system would be built with the intention of replacing Hug's current `direct
 ## Proposed Dependency Injection System
 
 This HOPE proposes a new dependency injection system with an aim for clarity and reuse.
-The API of the new system would consist of a `provides` decorator for defining available dependencies, and a `requires` function for using them:
+The API of the new system would consist of a `provide` decorator for defining available dependencies, and a `requires` function for using them:
```

Comment:
Require hints towards active voice, whereas requires hints towards passive. e.g. "It requires" vs "I require".

Reply:
I was talking about the `provide` vs `provides` more than the use (require) in this case but I can see the argument being similar. That being said I think
```python
@hug.provide('identifier')
def function(...
```
is a little bit more open to interpretation to what is doing the action. Is Hug providing something to the decorated function or is the decorated function providing something to the API?
On the other hand
```python
@hug.provides("identifier")
def function(...
```
to  me reads more clearly as the function providing `identifier` to the API.

However as I stated in the first comment I don't have strong feelings on this point, I'm fine with either form.

Reply:
The default behavior is to use the wrapped function name, I think overriding default behavior should require a named kwarg, e.g.

```
@hug.provide(alias="heres_a_name")
def function(...)
...

Reply:
Fair enough, I had forgotten that `name=` was an optional parameter and that it would use the wrapped function's name if not provided. I'll need to think more on if I agree with requiring that it is a named kwarg. Regardless in the case where the decorator is setting up a dependency injection named from the function I agree `@hug.provide()` sounds a bit better. This change seems sensible to me.